### PR TITLE
feat(diag): add config file and auth source to clever diag

### LINKED ISSUE #594: Add some details in the `clever diag`
## Context

We encountered the following situation multiple times:

* The user sets `CLEVER_TOKEN` and `CLEVER_SECRET` in a bash/zsh profil
* The user forgets about this
* The user tried to `clever login` but the CLI still uses the env vars
* If the token from the env vars is revoked, it appears like the clever login fails

## TODO

We should add some informations in the `clever diag` output:

* The location of the config file
* The source of the tokens (env vars or config file)

### PR DESCRIPTION
**Pull request by hsablonniere**
Closes #594

### CODE REVIEW

File: src/commands/diag.js [hunk 1]
```diff
@@ -28,12 +28,16 @@ async function diag () {
...
+
   const isPackaged = (process.pkg != null);
   Logger.println('Packaged      ' + colors.green(isPackaged));
   Logger.println('Exec path     ' + colors.green(process.execPath));
+  Logger.println('Config file   ' + colors.green(conf.CONFIGURATION_FILE));
 
   Logger.println('User id       ' + colors.green(userId || 'Not connected'));
-  Logger.println('oAuth token   ' + colors.green(oauthToken));
+  Logger.println('oAuth token   ' + colors.green(authDetails.token || 'Not connected'));
+  Logger.println('Auth source   ' + colors.green(authDetails.source));
```
Comment (aurrelhebert) [inline src/commands/diag.js:40]: I would have put this information above or just below the config file path

File: src/commands/diag.js [hunk 2]
```diff
@@ -28,12 +28,16 @@ async function diag () {
...
   }
 
+  Logger.println('Shell         ' + colors.green(process.env.SHELL));
+
   const isPackaged = (process.pkg != null);
   Logger.println('Packaged      ' + colors.green(isPackaged));
   Logger.println('Exec path     ' + colors.green(process.execPath));
+  Logger.println('Config file   ' + colors.green(conf.CONFIGURATION_FILE));
 
   Logger.println('User id       ' + colors.green(userId || 'Not connected'));
```
Comment (aurrelhebert) [inline src/commands/diag.js:38]: In case the user is not connected, maybe we can hide the information below (the oAuthToken and the auth source)?

**Comment by aurrelhebert** [comment]
LGTM only small comment :)

File: src/commands/diag.js [hunk 3]
```diff
@@ -28,12 +28,16 @@ async function diag () {
...
-  Logger.println('oAuth token   ' + colors.green(oauthToken));
+  Logger.println('oAuth token   ' + colors.green(authDetails.token || 'Not connected'));
```
Comment (aurrelhebert) [inline src/commands/diag.js:39]: And maybe print `Not connected` in an other color?

File: src/commands/diag.js (hunk 1 above)
Comment (hsablonniere) [reply src/commands/diag.js:40]: Agreed! :+1:

File: src/commands/diag.js (hunk 2 above)
Comment (hsablonniere) [reply src/commands/diag.js:38]: Agreed! :+1:

File: src/commands/diag.js (hunk 3 above)
Comment (hsablonniere) [reply src/commands/diag.js:39]: Agreed! :+1:

**Review by aurrelhebert — APPROVED** [review]

**Pull request merged**